test/extensions/filters/http/proto_api_scrubber/filter_test.cc

```diff
@@ -259,6 +379,107 @@ TEST_F(ProtoApiScrubberPassThroughTest, StreamingMultipleMessageSingleBuffer) {
   checkSerializedData<CreateApiKeyRequest>(*request_data4, {request4});
 }
 
+using ProtoApiScrubberScrubbingTest = ProtoApiScrubberFilterTest;
+
+TEST_F(ProtoApiScrubberFilterTest, UnknownGrpcMethod) {
+  ProtoApiScrubberConfig config;
+  ASSERT_TRUE(reloadFilter(config).ok());
+
+  // Prepare request
+  TestRequestHeaderMapImpl req_headers =
+      TestRequestHeaderMapImpl{{":method", "POST"},
+                               {":path", "/some.nonexistent.Service/UnknownMethod"},
+                               {":scheme", "http"},
+                               {"content-type", "application/grpc"}};
+  CreateApiKeyRequest request = makeCreateApiKeyRequest();
+  Envoy::Buffer::InstancePtr request_data = Envoy::Grpc::Common::serializeToGrpcFrame(request);
+
+  // The headers check passes because content-type is application/grpc.
+  EXPECT_EQ(Envoy::Http::FilterHeadersStatus::Continue, filter_->decodeHeaders(req_headers, true));
+
+  std::string expected_error_msg =
+      "Unable to find method `/some.nonexistent.Service/UnknownMethod` in the "
+      "descriptor pool configured for this filter.";
+
+  EXPECT_CALL(mock_decoder_callbacks_,
+              sendLocalReply(Envoy::Http::Code::BadRequest, Eq(expected_error_msg), _,
+                             Eq(Envoy::Grpc::Status::InvalidArgument),
+                             Eq("proto_api_scrubber_INVALID_ARGUMENT{BAD_REQUEST}")));
+
+  EXPECT_EQ(Envoy::Http::FilterDataStatus::StopIterationNoBuffer,
+            filter_->decodeData(*request_data, true));
+}
+
+TEST_F(ProtoApiScrubberScrubbingTest, ScrubRequestFieldBasedOnConfig) {
+  {
```

Comment:
Having multiple non-simple tests in one test are more challenging to maintain as one may not have the full context of the relationship between the different sub-tests. Is there a reason these are all part of the same "ScrubRequestFieldBasedOnConfig" test?
In any case, please add a one-line comment before each test that explains what it validates (the motivation).